### ISSUE TITLE
gpg: connecting dirmngr failed installing az command line tool on Ubuntu WSL

### ISSUE THREAD

**Issue by MaurGi** [Microsoft/WSL#3696]
This bug-tracker is monitored by developers and other technical types.  We like detail!  So please use this form and tell us, concisely but precisely, what's up.  Please fill out ALL THE FIELDS!  Issues with missing or incomplete issue templates will be closed.

If you have a feature request, please post to [the UserVoice](https://wpdev.uservoice.com/forums/266908).

If this is a console issue (a problem with layout, rendering, colors, etc.), please post to the [console issue tracker](https://github.com/microsoft/console/issues).

**Important: Do not open GitHub issues for Windows crashes (BSODs) or security issues.**  Please direct all Windows crashes and security issues to secure@microsoft.com.  Ideally, please [configure your machine to capture minidumps](https://support.microsoft.com/en-us/help/315263/how-to-read-the-small-memory-dump-file-that-is-created-by-windows-if-a), repro the issue, and send the minidump from "C:\Windows\minidump\".

**Please fill out the below information:**
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)

Microsoft Windows [Version 10.0.17134.345]

* What you're doing and what's happening:  (Copy&paste specific commands and their output, or include screen shots)

Following instructions to install az bash command line tool at `https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest#signingKey`
Running:

```
sudo apt-key --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg adv \
     --keyserver packages.microsoft.com \
     --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
```
I get:

```
Executing: /tmp/apt-key-gpghome.UioVyrUqCv/gpg.1.sh --keyserver packages.microsoft.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
gpg: connecting dirmngr at '/tmp/apt-key-gpghome.UioVyrUqCv/S.dirmngr' failed: IPC connect call failed
gpg: keyserver receive failed: No dirmngr
```
Ubuntu version: 
Description:    Ubuntu 18.04.1 LTS

* What's wrong / what should be happening instead:

expected result (works on Ubuntu vm):
```
Executing: /tmp/apt-key-gpghome.4GJTfoZeER/gpg.1.sh --keyserver packages.microsoft.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
gpg: key EB3E94ADBE1229CF: public key "Microsoft (Release signing) <gpgsecurity@microsoft.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
```

Ubuntu version: 
Description:    Ubuntu 18.04.1 LTS

* Strace of the failing command, if applicable:  (If `some_command` is failing, then run `strace -o some_command.strace -f some_command some_args`, and link the contents of `some_command.strace` in a [gist](https://gist.github.com/) here).

* For WSL launch issues, please [collect detailed logs](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs).

See [our contributing instructions](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md) for assistance.

**Comment by therealkenc** [comment]
#3286

**Comment by csmithson12345** [comment]
[FIXED!]
Heya folks, i have been banging my head on this for a few days off and on. 
i may be re-stating what is obvious to people, but this is what i determined.
The actual site packages.microsoft.com appears to be the issue. I dont know what mechanism in ubuntu server/workstation performs an update of trusted certificates, but i am assuming its not in place in WSL.

To fix this i went to packages.microsoft.com. I examined the certificate for that website. 
![image](https://user-images.githubusercontent.com/31697222/52723805-33061e80-2f7c-11e9-8d2d-4add1643985d.png)

I saved off each layer of the certificate to a CRT file, then copied those CRT files to a location within WSL /usr/local/share/ca-certificates/azure-cli 
(i created the azure-cli folder)
afterwords i updated the certificate list with the command
`sudo update-ca-certificates`
which resulted in the below output.
Updating certificates in /etc/ssl/certs...
4 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.

I can now install azure-cli in 18.04 ubuntu running on WSL.
